packages/rest/src/coercion/validator.ts

```diff
@@ -62,52 +56,18 @@ export class Validator {
    * @param opts
    */
   isRequired(opts?: ValidationOptions) {
-    if (this.ctx && this.ctx.schema && this.ctx.schema.required) return true;
+    if (this.ctx.parameterSpec.required) return true;
     if (opts && opts.required) return true;
     return false;
   }
 
-  /**
-   * The validation executed after type coercion. Like
-   * checking invalid values.
-   *
-   * @param type A parameter's type.
-   * @param value A parameter's raw value from http request.
-   * @param opts options.
-   */
-
-  validateParamAfterCoercion(
-    type: string,
-    value: any,
-    opts?: ValidationOptions,
-  ) {
-    switch (type) {
-      case 'number':
-        this.validateNumber(value);
-        break;
-      //@jannyhou: other types TBD
-      default:
-        return;
-    }
-  }
-
   /**
    * Return `true` if the value is empty, return `false` otherwise.
    *
    * @param value
    */
+  // tslint:disable-next-line:no-any
   isAbsent(value: any) {
-    const isEmptySet = [''];
-    return isEmptySet.includes(value);
-  }
-
-  /**
-   * Validate the coerced value for a number type parameter.
-   *
-   * @param value the coerced value of a number type parameter.
-   */
-  validateNumber(value: any) {
-    if (value === undefined) return;
-    if (isNaN(value)) throw new HttpErrors['400']();
+    return [''].includes(value);
```

Comment:
I find this weirdly complex, `value === ''` is easier to comprehend and faster at runtime.

Should we handle the case where the value was not provided at all (e.g. the request header is missing)? 

```ts
return value == null || value === '';
```

Reply:
@bajtos After the raw data converted by http client, it's always `''`, see test case https://github.com/strongloop/loopback-next/pull/1370/files#diff-66e4e0e78968649919599eb83a809326R51, so `null` and `''` are all converted to `''`.

> value === '' is easier to comprehend and faster at runtime.
👍

Reply:
I think your test case is not good enough. I am talking about the following situation:

- A parameter called `X-Correlation-Id` is defined as `{in: 'header', schema: {type: 'number'}}`
- The HTTP request does not contain any `X-Correlation-Id` header.

Please add a test for this edge case and make sure it's handled correctly.

I am fine with leaving this test case out of scope of this initial pull request as long as it's done as part of the user story you are working on.